modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2ResultSetIterator.java

```diff
@@ -162,6 +163,9 @@ private boolean fetchPage() throws IgniteCheckedException {
                     return false;
             }
             catch (SQLException e) {
+                if (X.hasCause(e, QueryMemoryTracker.TrackerWasClosedException.class) && canceled)
```

Comment:
This looks non optimal.
Let's make stacktrace analyse as a second condition.
if (canceled && X.hasCause...)

Also this conditions can be checked after 'errorCode' check.